### PR TITLE
[SC-935] Feature/asm

### PR DESCRIPTION
**Pull request by ZumZoom**
getter: -470 gas on 5 points
whitelist: -487 gas on 5 resolvers
total: 957 gas

### CODE REVIEW

**Comment by ZumZoom** [comment]
getter: -1292 gas on 5 points
whitelist: -917 gas on 5 resolvers

File: contracts/SettlementExtension.sol [hunk 1]
```diff
@@ -77,34 +77,63 @@ contract SettlementExtension is IPostInteraction, IAmountGetter, FeeBankCharger
...
-
-            if (block.timestamp <= auctionStartTime) {
-                return initialRateBump;
-            } else if (block.timestamp >= auctionFinishTime) {
-                return 0; // Means 0% bump
+    function _getRateBump(bytes calldata auctionDetails) private view returns (uint256 result) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly ("memory-safe") {
+            function timeWeightedAvg(t1, v1, t2, v2) -> avg {
+                avg := div(add(mul(sub(timestamp(), t1), v2), mul(sub(t2, timestamp()), v1)), sub(t2, t1))
```
Comment (SevenSwen) [inline contracts/SettlementExtension.sol:84]:
```suggestion
                avg := div(add(mul(sub(timestamp(), t1), v2), mul(sub(t2, timestamp()), v1)), sub(t2, t1)) // ((block.timestamp - t1) * v2 + (t2 - block.timestamp) * v1) / (t2 - t1)
```

Comment (ZumZoom) [reply contracts/SettlementExtension.sol:84]: It is not used in fact. I forgot to remove the `timeWeightedAvg` function 😩

**Comment by ZumZoom** [comment]
Closing as the gas gains are not significant.

**Pull request closed without merging**